src/m365/aad/commands/user/user-set.spec.ts

```diff
@@ -291,4 +388,70 @@ describe(commands.USER_SET, () => {
     } as any);
     assert(loggerLogSpy.notCalled);
   });
+
+  it('updates Azure AD user and set its manager by id', async () => {
+    sinon.stub(request, 'patch').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${formatting.encodeQueryParameter(userPrincipalName)}`) {
+        return userResponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    const putSpy = sinon.stub(request, 'put').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${userPrincipalName}/manager/$ref`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { userPrincipalName: userPrincipalName, displayName: displayName, managerUserId: managerUserId } });
+    assert.deepEqual(putSpy.lastCall.args[0].data, {
+      '@odata.id': 'https://graph.microsoft.com/v1.0/users/f4099688-dd3f-4a55-a9f5-ddd7417c227a'
+    });
+  });
+
+  it('updates Azure AD user and set its manager by user principal name', async () => {
+    sinon.stub(request, 'patch').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${formatting.encodeQueryParameter(userPrincipalName)}`) {
+        return userResponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    const putSpy = sinon.stub(request, 'put').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${userPrincipalName}/manager/$ref`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { verbose: true, userPrincipalName: userPrincipalName, displayName: displayName, managerUserName: managerUserName } });
+    assert.deepEqual(putSpy.lastCall.args[0].data, {
+      '@odata.id': 'https://graph.microsoft.com/v1.0/users/doe@contoso.com'
+    });
+  });
+
+  it('updates Azure AD user and removes manager', async () => {
+    sinon.stub(request, 'patch').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${formatting.encodeQueryParameter(userPrincipalName)}`) {
+        return userResponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${userPrincipalName}/manager/$ref`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { verbose: true, userPrincipalName: userPrincipalName, displayName: displayName, removeManger: true } });
```

Comment:
Let's add a check that the delete request was actually fired. Right now it's perfectly possible that we never execute a delete request in the code, the test would still succeed.